### DISCUSSION TITLE
Issue with S3 - not loading url correctly

### DISCUSSION THREAD

**Discussion by kcoax** [outline/outline#2884] [General]
Hello, 
I'm running the second installation of outline wiki self-hosted, and I'm still having issues with S3. 
User is created, with correct access policy, an S3 bucket is created, access point too, and CORS policy is edited to match domain etc. 

ENV file is updated, with correct AWS credentials, but when I try to upload any image it's giving me this error
(When creating document or changing user logo etc)

> {"error":"Inaccessible host: `s3'. This service may not be available in the `eu-central-1' region.","stack":"UnknownEndpoint: Inaccessible host: `s3'. This service may not be available in the `eu-central-1' region.\n    at Request.ENOTFOUND_ERROR (/opt/outline/node_modules/aws-sdk/lib/event_listeners.js:507:46)\n    at Request.callListeners (/opt/outline/node_modules/aws-sdk/lib/sequential_executor.js:106:20)\n    at Request.emit (/opt/outline/node_modules/aws-sdk/lib/sequential_executor.js:78:10)\n    at Request.emit (/opt/outline/node_modules/aws-sdk/lib/request.js:688:14)\n    at ClientRequest.error (/opt/outline/node_modules/aws-sdk/lib/event_listeners.js:339:22)\n    at ClientRequest.<anonymous> (/opt/outline/node_modules/aws-sdk/lib/http/node.js:96:19)\n    at ClientRequest.emit (events.js:400:28)\n    at ClientRequest.emit (domain.js:475:12)\n    at TLSSocket.socketErrorListener (_http_client.js:475:9)\n    at TLSSocket.emit (events.js:400:28)\n    at TLSSocket.emit (domain.js:475:12)\n    at emitErrorNT (internal/streams/destroy.js:106:8)\n    at emitErrorCloseNT (internal/streams/destroy.js:74:3)\n    at processTicksAndRejections (internal/process/task_queues.js:82:21)","level":"error","message":"Error uploading to S3 from URL"}

If I try changing the S3 URL, I will see different values in "Inaccessible host:... It seems it picks up just the first part, and don't care about the rest. 

But If I open my Outline wiki page, and check the source code, there however is the correct s3 URL loaded. 
However, I am still getting errors on trying to upload any image.

@tommoor Could you please advise? :)

**Comment by tommoor** [comment]
Try renaming the bucket to something random, it could be you're falling into this logic that is trying to detect the type of s3 host

https://github.com/outline/outline/blob/main/server/utils/s3.ts#L20-L22

**Comment by kcoax** [comment]
Hey @tommoor ,

thanks for reply! I will try, currently my aws env config looks like this

> AWS_ACCESS_KEY_ID=xxxx
AWS_SECRET_ACCESS_KEY=xxxxx
AWS_REGION=eu-central-1
AWS_S3_UPLOAD_BUCKET_URL=s3://arn:aws:s3:eu-central-1:xxxx:accesspoint/outline-wiki
AWS_S3_UPLOAD_BUCKET_NAME=outline-wiki
AWS_S3_UPLOAD_MAX_SIZE=26214400
AWS_S3_FORCE_PATH_STYLE=true
AWS_S3_ACL=public


And as you can see In the error its only gets 's3' part..Should protocol be s3 (as on AWS that it is s3 bucket URL looks like), or should there be HTTP protocol?

**Reply by tommoor** [reply]
You should use a regular url, eg: 

```
AWS_S3_UPLOAD_BUCKET_URL=https://my-host.s3.us-east-2.amazonaws.com
AWS_S3_UPLOAD_BUCKET_NAME=outline-attachments
```

**Reply by kcoax** [reply]
Thanks!
